python/cudf/cudf/core/column/column.py

```diff
@@ -1317,6 +1317,15 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
 
     elif np.isscalar(arbitrary) and not isinstance(arbitrary, memoryview):
         length = length or 1
+        if (
+            (nan_as_null is True)
+            and not is_string_dtype(arbitrary)
```

Comment:
I think this only checks if `arbitrary` is a string dtype object, not if `arbitrary` is possible a string scalar.

Reply:
We are already checking for it in the above condition `np.isscalar(arbitrary)`

Reply:
I mean that if arbitrary is `"abc"` this will return false as it's looking for either `"str"` or `np.dtype("object")` for example

Reply:
Ouch, missed that. `isinstance(arbitrary, str)` should be good I guess.

Reply:
Yea that should work I think.